torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -120,11 +120,10 @@ class TORCH_CUDA_CU_API FusionKernelRuntime {
   }
 
   // Try to compute heuristics based on the SegmentedFusion managed
-  //  in this kernel runtime, and will return a nullopt if either
-  //  any segment cannot be scheduled or the parameters don't match
-  using HeuristicsPtr = std::unique_ptr<FusionHeuristics>;
-  c10::optional<HeuristicsPtr> getMaybeHeuristicsFor(
-      const at::ArrayRef<IValue>& inputs);
+  //  in this kernel runtime. Returns true if a new set of heuristics
+  //  is successfully computed and the internally stored heuristics_ will
+  //  be updated with the new parameters.
+  bool getMaybeHeuristicsFor(const at::ArrayRef<IValue>& inputs);
```

Comment:
nit: the naming may be a little confusing as it implies heuristics to be returned

Reply:
Thanks for catching this. 👍  I missed the local commit that re-named this class and just separated it out.

Reply:
Updated. Thanks!